src/createBrowserLikeFetch.js

```diff
@@ -58,21 +60,29 @@ function createBrowserLikeFetch({
           cookieStrings.forEach((cookieString) => {
             const cookie = parse(cookieString);
             const { key, value: valueRaw, ...cookieOptions } = cookie.toJSON();
+            // first run tough-cookie's rules when adding the cookie to the jar
+            // if this throws then the cookie is not valid for the configured hostname either
             try {
-              const value = decodeURIComponent(valueRaw);
-              const cookieDomain = cookieOptions.domain;
-              if (cookieDomain && `.${cookieDomain}`.endsWith(`.${hostname.split('.').slice(-2).join('.')}`)) {
-                const expressCookieOptions = {
-                  ...cookieOptions,
-                  ...cookieOptions.maxAge ? { maxAge: cookieOptions.maxAge * 1e3 } : undefined,
-                };
-                res.cookie(key, value, expressCookieOptions);
-              }
               jar.setCookieSync(cookie, path);
             } catch (error) {
               // eslint-disable-next-line no-console
               console.warn(`Warning: failed to set cookie "${key}" from path "${path}" with the following error, "${error.message}"`);
+              return;
+            }
+
+            // then check if this cookie relates to this hostname
+            const cookieDomain = cookieOptions.domain;
+            if (!(cookieDomain && `.${cookieDomain}`.endsWith(dottedHostnamePublicSuffix))) {
```

Comment:
The cookie is not set if the domain is `localhost` which is annoying for testing locally but don't think we can allow localhost since is not a dotted suffix domain 😞